File: java/Consumer_Producer/Consumer_Producer/src/Buffer.java


public class Buffer<T> {
	T[] Buf;
	
	Buffer(int size){
		Buf =  (T[]) new Object[size];
	}
	
	public void einreihen(T a){
	boolean saved=false;
		for (int i = 0; i < Buf.length; i++) {
			if (Buf[i] == null && !saved) {
				Buf[i] = a;
				saved=true;
				notifyAll();
			}
		}
	}
	
	public T gibNextElement(){
		T tmp;
		for (int i = 0; i < Buf.length; i++) {
			if (Buf[i]!= null) {
				tmp = Buf[i];
				delElement();
				return tmp;
			}
		}
		return null;
	}
	
	public void delElement() {
		for (int i = 0; i < Buf.length-1; i++) {
			Buf[i] = Buf[i+1];
			Buf[i+1] = null;
		}
	}
	
	public int getSize(){
		return Buf.length;
	}

	public void print(){
		for (int i = 0; i < Buf.length; i++) {
			System.out.println(Buf[i]);
		}
	}

	public boolean istVoll(){
		boolean istVoll=true;
		for (int i = 0; i < Buf.length; i++) {
			if (Buf[i]==null) {
				istVoll=false;
			}
		}
		return istVoll;
	}
	
	public boolean istLeer(){
		boolean istLeer=true;
		for (int i = 0; i < Buf.length; i++) {
			if (Buf[i]!=null) {
				istLeer=false;
			}
		}
		return istLeer;
	}

}

File: python/C_P_PY/SRC/Buffer.py
from setuptools.command.build_ext import if_dl
from _overlapped import NULL
import threading

class Buffer(object):
    lst = [None] * 1
    lock = threading.Lock()



    
    def __init__(self, size):
        self.lst =[None]*size
        
        
    def addElement(self, a):
        saved = False
        for index, elem in enumerate(self.lst):
            if elem == None and not saved:
             self.lst[index] = a
             saved=True
    
    def getNextElement(self):
        for index, elem in enumerate(self.lst):
            if elem != None:
                 tmp = self.lst[index]
                 self.delElement()
                 return tmp
            return None
    
    def delElement(self):
        for i in range(len(self.lst)-1):
            self.lst[i] = self.lst[i+1]
            self.lst[i+1] = None
            
    def isFull(self):
        full=True
        for index, elem in enumerate(self.lst):
         if self.lst[index] == None:
             full = False
        return full
    
    def isEmpty(self):
        empty=True
        for index, elem in enumerate(self.lst):
         if self.lst[index] != None:
             empty = False
        return empty
    
    def getSize(self):
        return len(self.lst)
  